sim/core/sim.go

```diff
@@ -268,10 +301,13 @@ func (sim *Simulation) AddPendingAction(pa *PendingAction) {
 func (sim *Simulation) advance(elapsedTime time.Duration) {
 	sim.CurrentTime += elapsedTime
 
-	if !sim.executePhase && sim.CurrentTime >= sim.Encounter.executePhaseBegins {
-		sim.executePhase = true
-		for _, callback := range sim.executePhaseCallbacks {
-			callback(sim)
+	if !sim.executePhase {
+		if (sim.Encounter.EndFightAtHealth == 0 && sim.CurrentTime >= sim.Encounter.executePhaseBegins) ||
+			(sim.Encounter.EndFightAtHealth > 0 && sim.Encounter.EndFightAtHealth < sim.Encounter.Targets[0].DamageTaken) {
```

Comment:
This should check for 20% health remaining, not 0

Reply:
good catch